server/server/src/cold_chain/temperature_log.rs

```diff
@@ -0,0 +1,143 @@
+use actix_web::{
+    http::header,
+    web::{self, Data},
+    HttpRequest, HttpResponse,
+};
+use chrono::NaiveDateTime;
+use log::error;
+use mime_guess::mime;
+use repository::RepositoryError;
+use service::{
+    auth_data::AuthData,
+    service_provider::ServiceProvider,
+    temperature_log::{insert::InsertTemperatureLog, update::UpdateTemperatureLog},
+};
+
+use super::validate_request;
+
+#[derive(serde::Deserialize)]
+pub struct TemperatureLog {
+    id: String,
+    temperature: f64,
+    timestamp: i64,
+    #[serde(rename = "sensorId")]
+    sensor_id: String,
+    #[serde(rename = "temperatureBreachId")]
+    pub temperature_breach_id: Option<String>,
+}
+
+pub async fn put_logs(
+    request: HttpRequest,
+    service_provider: Data<ServiceProvider>,
+    auth_data: Data<AuthData>,
+    logs: web::Json<Vec<TemperatureLog>>,
+) -> HttpResponse {
+    let store_id = match validate_request(request, &service_provider, &auth_data) {
+        Ok((_user, store_id)) => store_id,
+        Err(error) => {
+            let formatted_error = format!("{:#?}", error);
+            return HttpResponse::Unauthorized().body(formatted_error);
+        }
+    };
+    let logs = logs.into_inner();
+    if !validate_input(&logs) {
+        return HttpResponse::BadRequest()
+            .body("Expecting a body with the array of temperature logs");
+    };
+
+    let results = match upsert_temperature_logs(service_provider, store_id, &logs).await {
+        Ok(response) => response,
+        Err(error) => return HttpResponse::InternalServerError().body(format!("{:#?}", error)),
+    };
+
+    HttpResponse::Ok()
+        .append_header(header::ContentType(mime::APPLICATION_JSON))
+        .json(&results)
+}
+
+fn validate_input(logs: &Vec<TemperatureLog>) -> bool {
+    logs.iter().all(|log| validate_log(log))
+}
+
+fn validate_log(log: &TemperatureLog) -> bool {
+    if log.timestamp < 1 {
+        return false;
+    }
+    true
+}
+
+async fn upsert_temperature_logs(
+    service_provider: Data<ServiceProvider>,
+    store_id: String,
+    logs: &Vec<TemperatureLog>,
+) -> Result<Vec<Result<repository::TemperatureLog, String>>, RepositoryError> {
+    let mut ctx = service_provider.basic_context()?;
```

Comment:
There was [this comment in previous PR](https://github.com/openmsupply/open-msupply/pull/2376#discussion_r1363134619), I think a part of it was missed, probably because I didn't emphasise it

Reply:
https://github.com/openmsupply/open-msupply/compare/2361-api-temperature-logs...2361-api-temperature-logs(Create-context-with-store-id)

Reply:
thanks - updated now